### ISSUE TITLE
`blacklisted_name` lint should not run inside the test code

### ISSUE THREAD

**Issue by popzxc** [rust-lang/rust-clippy#7305]
Lint name: `blacklisted_name`

I tried this code:

```rust
#[test]
fn test() {
   let foo_encoded = encode_object("foo"); // Actual functions are not important.
   let foo = parse_object(foo_encoded);
   assert_eq!(foo, "foo");
}
```

I expected to see this happen: `blacklisted_name` lint is not emitted. Foo is an OK name for tests, displaying something that is not really important as an object itself and telling programmers to focus on other things.

Instead, this happened: `blacklisted_name` lint is emitted.

### Meta

- `cargo clippy -V`: clippy 0.1.52 (9bc8c42b 2021-05-09)
- `rustc -Vv`:

  ```
  rustc 1.52.1 (9bc8c42bb 2021-05-09)
  binary: rustc
  commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
  commit-date: 2021-05-09
  host: aarch64-apple-darwin
  release: 1.52.1
  LLVM version: 12.0.0
  ```

**Comment by Diggsey** [comment]
Yeah this is going to create false positives in almost all crates with tests :|
